### PR TITLE
Temperature Sensor Driver

### PR DESCRIPTION
**Pull request by hana-t**

### CODE REVIEW

File: drivers/TempSensor.h
Comment (c-ola) [inline drivers/TempSensor.h:1]: Needs to  `#include <cstdint>` if you want to use the types `uint8_t` and `uint16_t` (I was getting errors trying to compile).

File: drivers/TempSensor.cpp
Comment (c-ola) [inline drivers/TempSensor.cpp:1]: Shouldn't have a main method, could move it to usage.cpp as a main method.

File: drivers/usage.cpp
Comment (c-ola) [inline drivers/usage.cpp:1]: This file is old? (no `ambientTemp()` method in TempSensor)

File: drivers/CMakeLists.txt
Comment (c-ola) [inline drivers/CMakeLists.txt:1]: You can have something like
```
add_executable(testTempSensor usage.cpp TempSensor.cpp)
target_link_libraries(testTempSensor i2c)
target_include_directories(testTempSensor PUBLIC .)
```
in the CMakeLists.txt for the tempsensor driver directory (also put the tempsensor files in its own directory.

The CMakeLists.txt should also add tempsensor.cpp  as a source to the main project target.
```
target_sources(${PROJECT_NAME} PUBLIC TempSensor.cpp)
target_include directories(${PROJECT_NAME} PUBLIC .)
``` 

Finally in the drivers CMakeLists.txt you just `add_subdirectory(tempsensor)`

**Review by c-ola — CHANGES_REQUESTED** [review]
Looks good, just needs some small organizational changes i think.

File: drivers/TempSensor.h
Comment (JaxHodg) [reply drivers/TempSensor.h:1]: Fixed, not sure why mine didn't error out

File: drivers/TempSensor.cpp
Comment (JaxHodg) [reply drivers/TempSensor.cpp:1]: Removed. That was previously used for testing

File: drivers/usage.cpp
Comment (JaxHodg) [reply drivers/usage.cpp:1]: Removed. That was previously used for testing

File: drivers/CMakeLists.txt
Comment (JaxHodg) [reply drivers/CMakeLists.txt:1]: Tbh I don't really understand cmake files, but the test should use `tests/Temperature_Sensor/test_TempSensor.cpp` not `usage.cpp`.

I've written `run_test.sh` just to get it quickly working

**Review by c-ola — APPROVED** [review]
looks good

**Review by HarrisonMKG — APPROVED** [review]
LGTM

**Pull request merged**